deploy/helm/flightctl/templates/flightctl-db-migration-job.yaml

```diff
@@ -1,164 +1,34 @@
-          valueFrom:
-            secretKeyRef:
-              name: flightctl-db-migration-secret
-              key: migrationUser
-        - name: DB_MIGRATION_PASSWORD
-          valueFrom:
-            secretKeyRef:
-              name: flightctl-db-migration-secret
-              key: migrationPassword
-        - name: DB_APP_USER
-          valueFrom:
-            secretKeyRef:
-              name: flightctl-db-app-secret
-              key: user
-        - name: DB_APP_PASSWORD
-          valueFrom:
-            secretKeyRef:
-              name: flightctl-db-app-secret
-              key: userPassword
-        - name: DB_ADMIN_USER
-          valueFrom:
-            secretKeyRef:
-              name: flightctl-db-admin-secret
-              key: masterUser
-        - name: DB_ADMIN_PASSWORD
-          valueFrom:
-            secretKeyRef:
-              name: flightctl-db-admin-secret
-              key: masterPassword
-        command:
-        - /bin/bash
-        - -c
-        - |
-          set -eo pipefail
-          echo "Running database migrations..."
-
-          # Copy config file to a writable location
-          mkdir -p /tmp/.flightctl
-          cp /root/.flightctl/config.yaml /tmp/.flightctl/config.yaml
-          export HOME=/tmp
-
-          /usr/local/bin/flightctl-db-migrate
-          echo "Migrations completed successfully!"
-
-          # Grant permissions on all existing tables to the application user
-          echo "Granting permissions on existing tables to application user..."
-          # Need to get admin credentials from init container environment
-          DB_HOST="{{ include "flightctl.dbHostname" . }}"
-          # Get admin credentials from the same secrets used by init container
-          export PGPASSWORD="$DB_ADMIN_PASSWORD"
-          psql -h "$DB_HOST" -p {{ (default 5432 $.Values.db.port) }} -U "$DB_ADMIN_USER" -d "{{ (default "flightctl" $.Values.db.name) }}" -c "SELECT grant_app_permissions_on_existing_tables();"
-          echo "Permission granting completed successfully!"
-        volumeMounts:
-        - mountPath: /root/.flightctl/
-          name: flightctl-db-migration-config
-          readOnly: true
-      volumes:
-      - name: flightctl-db-migration-config
-        configMap:
-          name: flightctl-db-migration-config
+{{- $root := . -}}
+{{- $baseName := printf "flightctl-db-migration-%d" .Release.Revision -}}
+
+{{- /* Defaults applied if any key along the path is missing */ -}}
+{{- $dbSetup := default (dict) .Values.dbSetup -}}
+{{- $migration := default (dict) $dbSetup.migration -}}
+{{- $dryRun := default (dict) $migration.dryRun -}}
+{{- $preUpgrade := default (dict) $dryRun.preUpgrade -}}
+
+{{- $dryRunEnabled := default true  $preUpgrade.enabled -}}
+{{- $dryRunHookWeight := default "-10" $preUpgrade.hookWeight -}}
+{{- $backoffLimit := default 3     $migration.backoffLimit -}}
+{{- $activeDeadlineSeconds := default 600   $migration.activeDeadlineSeconds -}}
```

Comment:
the defaults are already in `values.yaml`. Why default here again ? It will be a problem to keep these values in-sync as well.